rs/execution_environment/tests/dts.rs

```diff
@@ -351,20 +352,65 @@ fn dts_install_code_with_concurrent_ingress_sufficient_cycles() {
         return;
     }
     let install_code_ingress_cost = Cycles::new(INSTALL_CODE_INGRESS_COST);
+    let install_code_execution_cost = Cycles::new(INSTALL_CODE_EXECUTION_COST);
     let normal_ingress_cost = Cycles::new(NORMAL_INGRESS_COST);
     let max_execution_cost = Cycles::new(MAX_EXECUTION_COST);
     let actual_execution_cost = Cycles::new(ACTUAL_EXECUTION_COST);
 
     // The initial balance is sufficient to run `install_code` and to send an
     // ingress message concurrently.
-    let initial_balance = install_code_ingress_cost + normal_ingress_cost + max_execution_cost;
+    let initial_balance = install_code_ingress_cost
+        + install_code_execution_cost
+        + install_code_ingress_cost
+        + normal_ingress_cost
+        + max_execution_cost;
 
-    let DtsInstallCode {
-        env,
-        canister_id,
-        install_code_ingress_id,
-        config,
-    } = setup_dts_install_code(initial_balance, 0);
+    let (env, config) = dts_install_code_env(
+        NumInstructions::from(1_000_000),
+        NumInstructions::from(1000),
+    );
+
+    let canister_id = env.create_canister_with_cycles(
+        None,
+        initial_balance,
+        Some(
+            CanisterSettingsArgsBuilder::new()
+                .with_compute_allocation(1)
+                .with_freezing_threshold(0)
+                .build(),
+        ),
+    );
+
+    env.execute_ingress_as(
+        PrincipalId::new_anonymous(),
+        IC_00,
+        Method::InstallCode,
+        InstallCodeArgs::new(
+            CanisterInstallMode::Install,
+            canister_id,
+            wat::parse_str(DTS_INSTALL_WAT).unwrap(),
+            vec![],
+            None,
+            None,
+        )
+        .encode(),
+    )
+    .unwrap();
+
+    let install_code_ingress_id = env.send_ingress(
```

Comment:
Why do we need to install the canister first?

Reply:
Ingress messages to empty canisters are rejected by the ingress filter before they are enqueued. Hence, we need to make the canister non-empty so that the ingress message passes the ingress filter and gets enqueued for processing after the DTS completes.